bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -85,6 +85,22 @@ def run_plan(scope, plan_name, named_args = {})
       executor.report_yaml_plan(closure.model.body)
     end
 
+    # If a TargetSpec parameter is passed, ensure it is in inventory
+    inventory = Puppet.lookup(:bolt_inventory)
+
+    if inventory.version > 1
```

Comment:
Can we pull this algorithm into it's own function and use it in `run_task` as well, for #1175